src/main/controller/WorkoutController.ts

```diff
@@ -0,0 +1,39 @@
+import { Request, Response } from 'express';
+import db from '../models';
+import { Route } from '../constant/Route';
+
+export class WorkoutController {
+  // (9 - CC & WP) Retrieve detailed information about workout (title, duration, video link), exercises and their order, and info regarding exercises
+  getWorkoutDetails = async (req: Request, res: Response): Promise<any> => {
+    try {
+      if (!req.params.workoutID) {
+        throw `Missing workoutId param. usage: /${Route.WORKOUTS}/:workoutID(\\b[0-9a-f]{8}\\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\\b[0-9a-f]{12}\\b)`;
+      }
+
+      // TODO: make sure the user has permissions to view this workout (userId check? private / public check?)
+      const workouts = await db.Workout.findOne({
+        where: {
+          id: req.params.workoutID,
+        },
+        attributes: ['name', 'totalWorkoutTime', 'imageUrl'],
+        include: [
+          {
+            model: db.Exercise,
```

Comment:
I think we'll also need joins for the associated Exercise tables (for the tags). Likewise for the other endpoint! 

You can reference all the joins here - [Database Schema](https://www.notion.so/ubclaunchpad/Database-Schema-a9062419c1554fd59b358a57ea11b3ba)